### PR TITLE
*: support region split configuration.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @BusyJay @disksing @zhangjinpeng1987

### CODE REVIEW

File: src/raftstore/store/config.rs [hunk 1]
```diff
@@ -102,6 +102,18 @@ impl Config {
                                 self.raft_log_gc_threshold));
         }
 
+        if self.region_max_size <= self.region_split_size {
```
Comment (BusyJay) [inline src/raftstore/store/config.rs]: I think `region_max_size` can be equal to `region_split_size`.

File: src/raftstore/store/config.rs [hunk 2]
```diff
@@ -102,6 +102,18 @@ impl Config {
...
+            return Err(box_err!("region max size {} must > split size {}",
+                                self.region_max_size,
+                                self.region_split_size));
+        }
+
+        if self.region_check_size_diff >= self.region_split_size {
```
Comment (BusyJay) [inline src/raftstore/store/config.rs]: I don't think `region_check_size_diff` need to be larger than `region_split_size`.

Comment (siddontang) [inline src/raftstore/store/config.rs]: so remove this check and allow size diff > split size?

Comment (BusyJay) [inline src/raftstore/store/config.rs]: Yes, my mistake. What I actually wanted to say was `region_check_size_diff` can be larger than `region_split_size`.

**Comment by BusyJay** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

**Pull request merged**